### PR TITLE
Server can control multiple publisher placements

### PR DESCRIPTION
**Pull request by davidfischer**
This is based on #782 so as not to have conflicting migrations.

This change allows the server to control whether a publisher is allowed to have multiple placements (which the client sends since https://github.com/readthedocs/ethical-ad-client/pull/170). When merged, this will automatically reject multiple placements and we will need to turn it on per publisher.

### CODE REVIEW

File: adserver/api/serializers.py [hunk 1]
```diff
@@ -60,6 +60,10 @@ class AdDecisionSerializer(serializers.Serializer):
     # This purposefully doesn't use a URLField so we can disregard invalid values rather than rejecting them
     url = serializers.CharField(max_length=256, required=False)
 
+    # The placement index (0-indexed)
+    # 1 or more means there's multiple placements on this page
+    index = serializers.IntegerField(required=False, min_value=0, max_value=9)
```
Comment (ericholscher) [inline adserver/api/serializers.py:65]: Can we not have a max here? It seems unlikely, but I know the PSF has a use case where they're showing more than 9 images in a page.

File: adserver/api/views.py [hunk 2]
```diff
@@ -296,6 +296,7 @@ def decision(self, request, data):
                 keywords=keywords,
                 campaign_types=serializer.validated_data.get("campaign_types"),
                 url=url,
+                placement_index=serializer.validated_data.get("index"),
```
Comment (ericholscher) [inline adserver/api/views.py:299]: Should we call the incoming variable `placement_index` as well to keep it consistent?

**Review by ericholscher — APPROVED** [review]

File: adserver/api/serializers.py (hunk 1 above)
Comment (davidfischer) [reply adserver/api/serializers.py:65]: Sure. I can remove it. I'd rather have a very large max (99 or even 999) than no max at all though if that makes sense though.

File: adserver/api/views.py (hunk 2 above)
Comment (davidfischer) [reply adserver/api/views.py]: https://github.com/readthedocs/ethical-ad-client/pull/171

**Comment by davidfischer** [comment]
I updated based on the feedback.

**Comment by ericholscher** [comment]
Looks good 👍

**Pull request merged**